sdk/nodejs/runtime/closure/codePaths.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as fs from "fs";
+import * as filepath from "path";
+import * as readPackageTree from "read-package-tree";
+import * as asset from "../../asset";
+import { RunError } from "../../errors";
+
+// computeCodePaths computes the local node_module paths to include in an uploaded Lambda.
+export async function computeCodePaths(
```

Comment:
If we're exporting this, we should add a comment here indicating that you shouldn't need to use this unless you're writing code to interface to a provider's raw serverless primitives, which I imagine not many users will be doing.

Reply:
Sure.

Reply:
Done.